source/extensions/filters/http/oauth2/filter.cc

```diff
@@ -760,6 +760,12 @@ void OAuth2Filter::redirectToOAuthServer(Http::RequestHeaderMap& headers) {
   std::string csrf_token =
       Http::Utility::parseCookieValue(headers, config_->cookieNames().oauth_nonce_);
   bool csrf_token_cookie_exists = !csrf_token.empty();
+  // Validate the CSRF token HMAC if the CSRF token cookie exists.
```

Comment:
Can we add a brief comment explaining why we need to reset the CSRF token here? It would help future readers understand the reasoning behind this.